crates/interpreter/src/syntax.rs

```diff
@@ -546,6 +557,47 @@ impl CvtOp {
     }
 }
 
+#[cfg(feature = "threads")]
+macro_rules! impl_atomic_op {
+    ($u:ident) => {
+        paste::paste! {
+            impl AtomicUnOp {
+                pub fn $u(&self, mem: *mut $u, c: $u) -> $u {
+                    let atomic_val = unsafe { [< Atomic $u:upper >]::from_ptr(mem) };
```

Comment:
nit: can just use `x` as variable name. It's local and repetitive.

You can also rename `mem` to `ptr`.